### PR TITLE
network: fix configuring of CAN devices

### LINKED ISSUE #20428: [networkd] Regression in handling of socketcan interfaces after v248
**systemd version 249 (249.3-1-manjaro)**

This is my second, if not third, attempt to get `systemd-networkd` to manage CAN interface.
I do have a gs_usb enabled device which is properly detected by kernel. When I use standard commands listed below device works.
```
ip link set can0 type can bitrate 50000 restart-ms 100
ip link set can0 up
```
Yet, with same settings placed in network file it does not work.

**Used distribution**
 > `Manjaro / Arch`

**Linux kernel version used** (`uname -a`)
 > `Linux arch 5.13.9-1-MANJARO #1 SMP PREEMPT Sun Aug 8 09:02:03 UTC 2021 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > `x86_64`

**Expected behaviour you didn't see**
 > `Managed socketcan interface`

**Unexpected behaviour you saw**
 > `Interface does not come`

**Steps to reproduce the problem**
 > 1. Get a gs_usb dongle
 > 2. Create basic networkd configuration
 > 3. Plug dongle

**Additional program output to the terminal or log subsystem illustrating the issue**

Config file `/etc/systemd/network/80-can0.network`:
```text
[Match]
Name=*can0

[CAN]
BitRate=50000
RestartSec=100ms
```

Debug output from networkd
```text
can0: State changed: failed -> linger
can0: Permanent MAC address not found for new device, continuing without: Operation not supported
can0: Link 20 added
can0: Saved original MTU 16 (min: 0, max: 4294967295)
can0: Flags change: +NOARP +ECHO
can0: link pending udev initialization...
can0: udev initialized link
can0: State changed: pending -> initialized
can0: Link state is up-to-date
can0: found matching network '/etc/systemd/network/80-can0.network'.
can0: State changed: initialized -> configuring
can0: Requested to set CAN interface configurations
can0: Requested to activate link
can0: Setting CAN interface configurations
can0: Setting bitrate = 125000 bit/s
can0: Using default sample point
can0: Setting restart = 100ms
can0: Failed to set CAN interface configurations: No such device
can0: Failed
can0: State changed: configuring -> failed
```

I am not quite sure where from the "No such device" error comes from.

### PR DESCRIPTION
**Pull request by yuwata**
Fix a bug introduced by 7558f9e717381eef0ddc8ddfb5a754ea4b0f3e6c.

Fixes #20428.

### CODE REVIEW

**Comment by yuwata** [comment]
Needs to be backported to v249-stable.

File: src/network/networkd-setlink.c [hunk 1]
```diff
@@ -229,10 +229,14 @@ static int link_configure(
 
         log_link_debug(link, "Setting %s", set_link_operation_to_string(op));
 
-        if (IN_SET(op, SET_LINK_BOND, SET_LINK_CAN)) {
+        if (op == SET_LINK_BOND) {
                 r = sd_rtnl_message_new_link(link->manager->rtnl, &req, RTM_NEWLINK, link->master_ifindex);
                 if (r < 0)
                         return log_link_debug_errno(link, r, "Could not allocate RTM_NEWLINK message: %m");
+        } else if (op == SET_LINK_CAN) {
```
Comment (bluca) [inline src/network/networkd-setlink.c:236]: Isn't this new block identical to the else block below?

Comment (yuwata) [reply src/network/networkd-setlink.c:236]: No, netlink commands are different: RTM_NEWLINK vs RTM_SETLINK.

Comment (bluca) [reply src/network/networkd-setlink.c:236]: duh, and I checked 3 times

**Review by bluca — APPROVED** [review]
lgtm

**Pull request merged**